test/tools/OpenCover/OpenCover.psm1

```diff
@@ -1,113 +1,187 @@
-﻿
-#region Classes
-Class CoverageSummary {
-    [int]$NumSequencePoints
-    [int]$VisitedSequencePoints
-    [int]$NumBranchPoints
-    [int]$VisitedBranchPoints
-    [double]$SequenceCoverage
-    [double]$BranchCoverage
-    [int]$MaxCyclomaticComplexity
-    [int]$MinCyclomaticComplexity
-    [int]$VisitedClasses
-    [int]$NumClasses
-    [int]$VisitedMethods
-    [int]$NumMethods
-    CoverageSummary([xml.xmlelement]$x) {
-        $this.numSequencePoints = $x.numSequencePoints
-        $this.visitedSequencePoints = $x.visitedSequencePoints
-        $this.numBranchPoints = $x.numBranchPoints
-        $this.visitedBranchPoints = $x.visitedBranchPoints
-        $this.sequenceCoverage = $x.sequenceCoverage
-        $this.branchCoverage = $x.branchCoverage
-        $this.maxCyclomaticComplexity = $x.maxCyclomaticComplexity
-        $this.minCyclomaticComplexity = $x.minCyclomaticComplexity
-        $this.visitedClasses = $x.visitedClasses
-        $this.numClasses = $x.numClasses
-        $this.visitedMethods = $x.visitedMethods
-        $this.numMethods = $x.numMethods
-    }
-    [string]ToString() {
-        return "Branch:{0,3} Sequence:{1,3}" -f $this.BranchCoverage,$this.SequenceCoverage
-    }
+﻿#region privateFunctions
+
+function Get-AssemblyCoverageData([xml.xmlelement] $element)
+{
+    $AssemblyCoverageData = New-Object System.Management.Automation.PSObject
+    $AssemblyCoverageData | Add-Member -MemberType NoteProperty -Name AssemblyName -TypeName [string] -Value ([string]::Empty)
```

Comment:
The `-TypeName` isn't working how you think here - it's adding the literal string `[string]` to PSTypeNames on `$AssemblyCoverageData.